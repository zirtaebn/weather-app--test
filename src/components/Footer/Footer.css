footer {

    position: fixed;
    bottom: 15px;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

}

footer .languages {

    display: flex;
    justify-content: center;

}

.languages div {

    height: 20px;
    width: 20px;

    cursor: pointer;

    margin-right: 12px;
    background-repeat: no-repeat;
    background-size: cover; 
    background-position: bottom;

    
}

.languages div:last-child{

    
    margin-right: 0;
    
}

div.pt {

    background-image: url('../../assets/images/pt.png');
} 

div.en {

    background-image: url('../../assets/images/en.png');
}

div.es {

    background-image: url('../../assets/images/es.png');
}


footer p {

    margin-top: 5px;
    font-weight: 400;
    font-size: 10px;
    line-height: 12px;
    letter-spacing: 0.05em;
    color: #fff;
}