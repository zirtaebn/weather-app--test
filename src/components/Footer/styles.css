footer {

    position: fixed;
    bottom: 15px;

    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

footer p {

    margin-top: 3px;
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 400;
    font-size: 10px;
    line-height: 12px;
    letter-spacing: 0.05em;
    color: #fff;
}

footer .languages {

    display: flex;

}

.languages div {

    height: 20px;
    width: 20px;

    cursor: pointer;

    margin-right: 12px;

    
}

div.pt {

    background-image: url('../../images/pt.png');
} 

div.en {

    background-image: url('../../images/en.png');


}

div.es {

    background-image: url('../../images/es.png');


}