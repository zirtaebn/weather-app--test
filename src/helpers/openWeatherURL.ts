import { initialStateType } from '../contexts/Context';

export const openWeatherURL = (parameter: string, state: initialStateType) => {
    
    return `https://api.openweathermap.org/data/2.5/${parameter}?lat=${state.adress.lat}&lon=${state.adress.lng}&appid=${process.env.REACT_APP_WEATHER_API_KEY}&lang=${state.language.name}&units=${state.temp.isToggle ? 'metric' : 'imperial'}`;
}