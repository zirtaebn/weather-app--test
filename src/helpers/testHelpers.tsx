import { render } from '@testing-library/react';

import { IntlContext, initialStateType, IntlContextProvider } from '../contexts/IntlContext';

export const renderWithContext = (ui:React.ReactElement, state?: initialStateType, renderOptions?: any) => {
            
    return render(
        state ? <IntlContext.Provider value={{state} as any}>{ui}</IntlContext.Provider> : <IntlContextProvider>{ui}</IntlContextProvider>,
        renderOptions,
    )
} 