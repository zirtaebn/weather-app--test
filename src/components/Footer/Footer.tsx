import './Footer.css';

import { IntlContext } from '../../contexts/IntlContext';
import { usePersistedData } from '../../hooks/usePersistedData';
import { useLanguageString } from '../../hooks/useLanguageString';

import { useContext, useEffect } from 'react';

export const Footer = () => {

    const { state, dispatch } = useContext(IntlContext);
    const { footerMessage } = useLanguageString(state);
    const persistedData = usePersistedData('language',  state.language.name);
    
    useEffect(() => {
        
        dispatch({

            type: 'CHANGE_LANGUAGE',
            payload: {

                name: persistedData 
            }
        })
        // eslint-disable-next-line
    },[]);


    const handleChangeLanguage = (language: string) => {

        dispatch({

            type: 'CHANGE_LANGUAGE',
            payload: {

                name: language
            }
        })
    };

    
    return(

        <footer className="footer">
            <div className="footer__languages">
                <div
                    aria-label="change the language to portuguese"
                    onClick={() => handleChangeLanguage('pt')}
                >
                    
                </div>

                <div 
                    className="footer__language en"
                    aria-label="change the language to english"
                    onClick={() => handleChangeLanguage('en')}
                >  
                </div>

                <div
                    className="footer__language es"
                    aria-label="change the language to spanish"
                    onClick={() => handleChangeLanguage('es')}
                >

                </div>
            </div>
            
            <p className="footer__message">{ footerMessage }</p>
        </footer>


    )


}