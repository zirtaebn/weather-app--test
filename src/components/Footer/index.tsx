import './styles.css';

import { Context } from '../../contexts/Context';

import { usePersistedData } from '../../utils/usePersistedData';

import { useContext, useEffect } from 'react';



export const Footer = () => {

    const { state, dispatch } = useContext(Context);

    const persistedData = usePersistedData('language',  state.language.name);
    useEffect(() => {
        
        dispatch({

            type: 'CHANGE_LANGUAGE',
            payload: {

                name: persistedData 
            }
        })
        // eslint-disable-next-line
    },[]);


    let phrase: string = '';

    if(state.language.name === 'pt') {

        phrase = 'Idioma selecionado: Português.';

    }else if(state.language.name === 'en') {

        phrase = 'Selected language: English.';

    }else if(state.language.name === 'es'){

        phrase = 'Idioma seleccionado: Español.'
    }
    

    const handleChangeLanguage = (language: string) => {

        dispatch({

            type: 'CHANGE_LANGUAGE',
            payload: {

                name: language
            }
        })
    };

    
    return(

        <footer>
            <div className="languages">
                <div
                    className='pt'
                    onClick={() => {handleChangeLanguage('pt')}}
                >
                    
                </div>

                <div 
                    className='en'
                    onClick={() => {handleChangeLanguage('en')}}
                >  
                </div>

                <div 
                    className='es'
                    onClick={() => {handleChangeLanguage('es')}}
                >

                </div>
            </div>



            <p>{ phrase }</p>
        </footer>


    )


}