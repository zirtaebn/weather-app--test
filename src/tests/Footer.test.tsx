import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router-dom';
import { Footer } from '../components/Footer/Footer';
import { renderWithContext } from '../helpers/testHelpers';


describe('[COMPONENTS]: Footer', () => {
    it('it should render correctly', () => {
        renderWithContext(
          <MemoryRouter>
            <Footer />
          </MemoryRouter>,
        )
    
        expect(screen.getByLabelText('change the language to portuguese')).toMatchSnapshot()
    });
    
    it('it should display footer message in Portuguese', async () => {
        renderWithContext(
            <MemoryRouter>
              <Footer />
            </MemoryRouter>,
          )
          userEvent.click(screen.getByLabelText('change the language to portuguese'))

        expect(await screen.findByText('Idioma selecionado: Português.')).toMatchSnapshot()
    });
    
    it('it should display footer message in English', async () => {
        renderWithContext(
            <MemoryRouter>
              <Footer />
            </MemoryRouter>,
          )

        userEvent.click(screen.getByLabelText('change the language to english'))

        expect(await screen.findByText('Selected language: English.')).toMatchSnapshot()
    });
    
    it('it should display footer message in Spanish', async () => {
        renderWithContext(
            <MemoryRouter>
              <Footer />
            </MemoryRouter>,
          )
          userEvent.click(screen.getByLabelText('change the language to spanish'))

        expect(await screen.findByText('Idioma seleccionado: Español.')).toMatchSnapshot()
    });
})
