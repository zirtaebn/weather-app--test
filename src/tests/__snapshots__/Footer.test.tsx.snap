// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENTS]: Footer it should display footer message in English 1`] = `
<p
  class="footer__message"
>
  Selected language: English.
</p>
`;

exports[`[COMPONENTS]: Footer it should display footer message in Portuguese 1`] = `
<p
  class="footer__message"
>
  Idioma selecionado: Português.
</p>
`;

exports[`[COMPONENTS]: Footer it should display footer message in Spanish 1`] = `
<p
  class="footer__message"
>
  Idioma seleccionado: Español.
</p>
`;

exports[`[COMPONENTS]: Footer it should render correctly 1`] = `
<div
  aria-label="change the language to portuguese"
  class="footer__language pt"
  data-testid="pt-br-language-icon"
/>
`;
